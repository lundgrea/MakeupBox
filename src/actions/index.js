export const handleErrors = error => ({
  type: "HANDLE_ERROR",
  error
})

export const isLoading = bool => ({
  type: 'IS_LOADING',
  bool
})

export const getResults = (results) => ({
  type: 'GET_RESULTS',
  results
})

export const setResponses = response => ({
  type: 'SET_RESPONSES',
  response
})