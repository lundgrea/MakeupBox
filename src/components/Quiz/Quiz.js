import React, { Component } from 'react';
import './Quiz.css';
import QuizQuestion from '../../components/QuizQuestion/QuizQuestion';

export class Quiz extends Component {
  constructor() {
    super()
    this.state = {
      questionOneAnswer: '',
      questionTwoAnswer: '',
      questionThreeAnswer: '',
      questionFourAnswer: '',
      quizQuestions: [
        {
          question: 'What is your primary interest?', 
          answers: ['lips', 'eyes','face/body', 'nails'],
          id: 1
        },
        {
          question: 'What style speaks most to you?', 
          answers: ['bronzed','bold', 'natural', 'smoky'],
          id: 2
        },
        { 
          question: 'What are your causes?',
          answers: ['cruelty free', 'organic', 'non-gmo', 'vegan'],
          id: 3
        }, 
        {
          question: 'What is your previous experience using makeup',
          answers: ['none', 'some', 'a bit', 'absolute master'],
          id: 4
        }
      ]    
    }
  }

  generateQuiz = () => {
    return this.state.quizQuestions.map(question => {
      return (
        <form>
          <QuizQuestion question={question.question} answers={question.answers} key={question.id} />
        </form>
      )
    })    
  }


  
  render() {
    return (
      <fieldset>
        {this.generateQuiz()}
      </fieldset>
    )
}


}

export default Quiz;