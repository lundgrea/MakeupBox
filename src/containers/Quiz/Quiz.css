fieldset {
  text-align: center;
  max-width: 700px;
  margin: auto;
  margin-top: 30px;
  border-radius: 2px;
  border: none;
  color: #242B28;
  background-color: #8EA99E;
}

label {
  margin: 0px 0px 0px 10px;
  font-size: 22px;
}

.button-quiz-submit {
  border: none;
  margin: 10px 10px 10px 10px;
  background-color: #242B28;
  color: #8EA99E;
  border: none;
  border-radius: 2px;
  width: 45%;
  height: 40px;
  font-size: 20px;
}

.button-quiz-submit:hover {
  color: white;
}

.quiz-headline {
  font-size: 30px;
}
