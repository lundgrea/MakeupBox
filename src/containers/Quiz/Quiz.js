import React, { Component } from 'react';
import './Quiz.css';
import QuizQuestion from '../../components/QuizQuestion/QuizQuestion';
import { setResponses } from "../../actions";
import { connect } from 'react-redux';
import { Link } from 'react-router-dom'

export class Quiz extends Component {
  constructor() {
    super()
    this.state = {
      userAnswers: [],     
      results: [], 
      quizQuestions: [
        {
          question: 'What is your primary interest?', 
          answers: ['lips', 'eyes','face/body', 'nails'],
          id: 1,
          name: 'interest'
        },
        {
          question: 'What style speaks most to you?', 
          answers: ['bronzed','bold', 'natural', 'smokey'],
          id: 2,
          name: 'style'
        },
        { 
          question: 'What are your causes?',
          answers: ['cruelty free', 'organic', 'non-gmo', 'vegan'],
          id: 3,
          name: 'causes'
        }, 
        {
          question: 'What is your previous experience using makeup',
          answers: ['none', 'some', 'a bit', 'absolute master'],
          id: 4,
          name: 'experience'
        }
      ]    
    }
  }

  generateQuiz = () => {
    return this.state.quizQuestions.map(question => {
      return (
        <QuizQuestion question={question.question} answers={question.answers} key={question.id} id={question.id} name={question.name} handleChange={this.handleChange}/>
      )
    })    
  }

  handleChange = (e) => {
    this.setState({userAnswers: [...this.state.userAnswers, e.target.value]})
    this.props.setResponses(e.target.value)
  }

  render() {
    return (
      <fieldset>
        {this.generateQuiz()}
        <Link to='/results'><button>submit answers</button></Link>
      </fieldset>
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  setResponses: response => dispatch(setResponses(response))
});

export default connect(null, mapDispatchToProps)(Quiz);