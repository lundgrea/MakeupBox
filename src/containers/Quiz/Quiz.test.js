import React from 'react';
import { shallow } from 'enzyme';
import { Quiz, mapDispatchToProps } from './Quiz';
import { setResponses } from "../../actions";

describe('Quiz', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Quiz />);
    expect(wrapper).toMatchSnapshot();
  });

  // it('should update state with a the quiz responses when handleChange is called', () => {
  //   const wrapper = shallow(<Quiz /> );
  //   wrapper.instance().setResponses() = jest.fn()
  //   const mockEvent = {target: 'lips'}
  //   const expected = [mockEvent];
  //   expect(wrapper.state('userAnswers')).toEqual([]);
  //   wrapper.instance().handleChange(mockEvent);
  //   expect(wrapper.state('userAnswsers')).toEqual(expected)
  // })
});

describe('mapDispatchToProps', () => {
  it('should call dispatch with setResponses action', () => {
    const mockResponse = ['lips']
    const mockDispatch = jest.fn();
    const actionToDispatch = setResponses(mockResponse)
    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.setResponses(mockResponse)
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })
})

