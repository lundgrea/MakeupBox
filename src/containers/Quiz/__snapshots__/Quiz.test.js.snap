// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz should match the snapshot 1`] = `
<fieldset
  className="quiz-container"
>
  <h2
    className="quiz-headline"
  >
    Help Us Tailor Your Results To Suit Your Needs
  </h2>
  <QuizQuestion
    answers={
      Array [
        "lips",
        "eyes",
        "face/body",
        "nails",
      ]
    }
    handleChange={[Function]}
    id={1}
    key="1"
    name="interest"
    question="1. What is your primary area of interest?"
  />
  <QuizQuestion
    answers={
      Array [
        "bronzed",
        "bold",
        "natural",
        "smokey",
      ]
    }
    handleChange={[Function]}
    id={2}
    key="2"
    name="style"
    question="2. What style describes you best?"
  />
  <QuizQuestion
    answers={
      Array [
        "cruelty free",
        "organic",
        "non-gmo",
        "vegan",
      ]
    }
    handleChange={[Function]}
    id={3}
    key="3"
    name="causes"
    question="3. What causes do you care about?"
  />
  <QuizQuestion
    answers={
      Array [
        "none",
        "some",
        "a bit",
        "absolute master",
      ]
    }
    handleChange={[Function]}
    id={4}
    key="4"
    name="experience"
    question="4. What previous experience do you have using makeup?"
  />
  <Link
    to="/results"
  >
    <button
      className="button-quiz-submit"
    >
      submit answers
    </button>
  </Link>
</fieldset>
`;
